### PR TITLE
Adding deprecation warnings to the view package

### PR DESCRIPTION
**Pull request by burzum**
Refs #11385

### CODE REVIEW

File: src/View/Helper/HtmlHelper.php [hunk 1]
```diff
@@ -784,6 +799,11 @@ public function getCrumbList(array $options = [], $startText = false)
      */
     protected function _prepareCrumbs($startText, $escape = true)
     {
+        deprecationWarning(
+            'HtmlHelper::_prepareCrumbs() is deprecated. ' .
```
Comment (saeideng) [inline src/View/Helper/HtmlHelper.php]: even for `protected` methods ?

Comment (burzum) [reply src/View/Helper/HtmlHelper.php]: Well, you can call them? @markstory has not told me any restrictions and I don't think it hurts to have it in one more place?

Comment (inoas) [reply src/View/Helper/HtmlHelper.php]: Makes sense to deprecate those as well, as long as it is not `final private` ;-)...

Comment (saeideng) [reply src/View/Helper/HtmlHelper.php]: >Well, you can call them when extending the class
OK :+1:

**Comment by markstory** [comment]
Tests will need to be updated to either not use the deprecated methods or use the `deprecated()` test helper to change the error reporting level.

File: src/View/Helper/HtmlHelper.php (hunk 1 above)
Comment (markstory) [reply src/View/Helper/HtmlHelper.php]: We should warn on protected methods as well. They will be removed and break code in the future.

**Comment by markstory** [comment]
I think I got most of the tests fixed up.

**Comment by ADmad** [comment]
Few CS errors need ti be fixed.

**Pull request merged**